lib/BackbeatConsumer.js

```diff
@@ -535,6 +540,72 @@ class BackbeatConsumer extends EventEmitter {
         return undefined;
     }
 
+    /**
+     * Pauses consumption of assigned partitions
+     * @returns {undefined}
+     */
+    _pauseAssignments() {
+        const assignments = this._consumer?.assignments();
+        // removing offset from assignments
+        const topicPartitions = assignments?.map(({ topic, partition }) => ({ topic, partition }));
```

Comment:
nit: may be clearer by not using destructuring (the function looks like a "nop")

```suggestion
        const topicPartitions = assignments?.map(assignment => ({ topic: assignment.topic, partition: assignment.partition }));
```